### ISSUE TITLE
Layout/HeredocArgumentClosingParenthesis false positive when chaining methods that pass heredocs

### ISSUE THREAD

**Issue by pdobb** [rubocop-hq/rubocop#9161]
When an outer method contains a chained method call with heredocs present, the Layout/HeredocArgumentClosingParenthesis cop appears to produce a false positive offense.

--------

## Expected behavior

No offenses should be found.

## Actual behavior

```ruby
# Baseline case (works fine)
my_outer_method(
  MyObject
    .select(Arel.sql("..."))
    .where(Arel.sql("...")))

# Raises an offense when Strings are changed to heredocs
my_outer_method(
  MyObject
    .select(Arel.sql(<<~SQL.squish))
      ...
    SQL
    .where(Arel.sql(<<~SQL.squish)))
      ...
    SQL

###

temp.rb:15:36: C: Layout/HeredocArgumentClosingParenthesis: Put the closing parenthesis for a method call with a HEREDOC parameter on the same line as the HEREDOC opening. (https://rubystyle.guide#heredoc-argument-closing-parentheses)
    .where(Arel.sql(<<~SQL.squish)))
                                   ^
```

Also:

```ruby
# Also raises the same offense:
my_outer_method(-> {
  my_inner_method(<<~SQL.squish)
    ...
  SQL
})

####

rcs/temp.rb:7:2: C: Layout/HeredocArgumentClosingParenthesis: Put the closing parenthesis for a method call with a HEREDOC parameter on the same line as the HEREDOC opening. (https://rubystyle.guide#heredoc-argument-closing-parentheses)
})
 ^
```
## Steps to reproduce the problem

There is no special configuration for this cop. So just run

```bash
rubocop --only Layout/HeredocArgumentClosingParenthesis temp.rb
```
on
```ruby
my_outer_method(
  MyObject
    .select(Arel.sql(<<~SQL.squish))
      ...
    SQL
    .where(Arel.sql(<<~SQL.squish)))
      ...
    SQL
```

NOTE: In my testing, the code must contain an outer method as well (which adds in an additional closing paren) to get the offense. E.g. the following does *not* produce any offenses:

```ruby
# Works fine due to no outer method (I presume).
MyObject
  .select(Arel.sql(<<~SQL.squish))
    ...
  SQL
  .where(Arel.sql(<<~SQL.squish))
    ...
  SQL
```

UPDATE: This fails as well:

```ruby
my_outer_method(-> {
  my_inner_method(<<~SQL.squish)
    ...
  SQL
})
```

## RuboCop version

I've tested this on both 1.4.2 and 1.5.1. Both exhibit the above issue.

```
$ rubocop -V
1.4.2 (using Parser 2.7.2.0, rubocop-ast 1.2.0, running on ruby 2.6.6 x86_64-darwin19)
  - rubocop-performance 1.9.0
  - rubocop-rails 2.8.1
  - rubocop-rake 0.5.1

# AND

$ rubocop -V
1.5.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.6.6 x86_64-darwin19)
  - rubocop-performance 1.9.0
  - rubocop-rails 2.8.1
  - rubocop-rake 0.5.1
```

**Comment by pdobb** [comment]
Note, I edited and added a 2nd false positive example to the description, but adding here as well to be extra clear:

```ruby
my_outer_method(-> {
  my_inner_method(<<~SQL.squish)
    ...
  SQL
})
```

```
$ rubocop --only Layout/HeredocArgumentClosingParenthesis temp.rb

temp.rb:7:2: C: Layout/HeredocArgumentClosingParenthesis: Put the closing parenthesis for a method call with a HEREDOC parameter on the same line as the HEREDOC opening. (https://rubystyle.guide#heredoc-argument-closing-parentheses)
})
 ^
```
